### PR TITLE
Fixes an issue with duplicate packages in image resolution

### LINKED ISSUE #1858: Image Resolution Duplicate Package
**Describe the bug**

If I ask to have an image resolved with two identical packages, the resulting resolution also contains two identical packages.

**Environment (please complete the following information):**

 - OS: PoopOS
 - OpenTAP Version: 9.27.0

**To Reproduce**

1. Have Runner 1.3.1 in your local package cache `~/.local/share/OpenTap/PackageCache`

2. Run `./tap image install --target ../Testing "Runner:1.3.1,Runner:1.3.1" --dry-run -v --repository ~/.local/share/OpenTap/PackageCache`

**Actual behavior**

There are two Runner packages in the resulted image:

image

### PR DESCRIPTION
**Pull request by Denrask**
Fixes #1858

### CODE REVIEW

File: Package/Image/ImageResolver.cs [hunk 1]
```diff
@@ -37,6 +37,30 @@ public ImageResolution ResolveImage(ImageSpecifier image, PackageDependencyGraph
...
+                            if (pkg1.Version.MatchBehavior == VersionMatchBehavior.Exact && pkg2.Version.MatchBehavior == VersionMatchBehavior.Compatible)
+                            {
+                                packages.RemoveAt(j);
+                                goto retry;
+                            }
+                        
+                            if (pkg2.Version.MatchBehavior == VersionMatchBehavior.Exact && pkg1.Version.MatchBehavior == VersionMatchBehavior.Compatible)
+                            {
+                                packages.RemoveAt(i);
+                                goto retry;
```
Comment (alnlarsen) [inline Package/Image/ImageResolver.cs:60]: This looks like a bug. 

`pkg1` is assigned right before the `retry` label, and this goto will continue the inner loop without reassigning `pkg1`.

This should probably be:

```
packages[i] = packages[j];
packages.RemoveAt(j);
goto retry;
```

**Review by alnlarsen — DISMISSED** [review]
I am not sure if this is the correct place to solve the problem. Otherwise the logic looks fine except for the bug I mentioned.

It also seems a bit weird that our Version comparison implementation does not check the match behavior, and I don't like the fact that this implementation relies on that behavior.

File: Package/Image/ImageResolver.cs [hunk 2]
```diff
@@ -37,6 +37,30 @@ public ImageResolution ResolveImage(ImageSpecifier image, PackageDependencyGraph
...
+                            }
+                        }
+
```
Comment (alnlarsen) [inline Package/Image/ImageResolver.cs:63]: It doesn't look like this is needed at all?

**Review by rmadsen-ks — DISMISSED** [review]
Looks good!

**Pull request merged**